pkg/imgpkg/bundle/bundle.go

```diff
@@ -117,11 +117,35 @@ func (o *Bundle) NoteCopy(processedImages *imageset.ProcessedImages, reg ImagesM
 
 	logger.Debugf("creating Locations OCI Image\n")
 	// Using NewNoopUI because we do not want to have output from this push
-	err = NewLocations(logger).Save(reg, destinationRef, locationsCfg, goui.NewNoopUI())
+
+	locations := NewLocations(logger)
+	err = locations.Save(reg, destinationRef, locationsCfg, goui.NewNoopUI())
 	if err != nil {
+		return o.debugFriendlyError(locations, reg, destinationRef, err, locationsCfg)
+	}
+
+	return err
+}
+
+func (o *Bundle) debugFriendlyError(locations *LocationsConfigs, reg ImagesMetadataWriter, destinationRef regname.Digest, err error, locationsCfg ImageLocationsConfig) error {
+	existingLocationsCfg, debugLocationErr := locations.Fetch(reg, destinationRef)
+	if debugLocationErr != nil {
 		return err
 	}
-	return nil
+
+	attemptedLocationCfgUploadBytes, debugLocationErr := locationsCfg.AsBytes()
+	if debugLocationErr != nil {
+		return err
+	}
+	existingLocationCfgUploadBytes, debugLocationErr := existingLocationsCfg.AsBytes()
+	if debugLocationErr != nil {
+		return err
+	}
+
+	return fmt.Errorf(`Unable to write location oci: %s
+location oci already uploaded contents: %s
+attempted to write oci contents: %s`,
```

Comment:
nit: in general its handy to add some deliminator like quotes or >><< around unknown multi line text so that we now within the error if there is any suspicious whitespace etc.

Reply:
ok done